apps/chat/src/utils/auth/auth-providers.ts

```diff
@@ -9,174 +9,282 @@ import OktaProvider from 'next-auth/providers/okta';
-    !!process.env.AUTH_OKTA_ISSUER &&
-    OktaProvider({
-      clientId: process.env.AUTH_OKTA_CLIENT_ID,
-      clientSecret: process.env.AUTH_OKTA_CLIENT_SECRET,
-      issuer: process.env.AUTH_OKTA_ISSUER,
-      authorization: {
-        params: {
-          scope: process.env.AUTH_OKTA_SCOPE || 'openid email profile',
+        token: tokenConfig,
+      })
+    : null;
+
+const getCognitoProvider = (config: ProviderConfig) =>
+  config.clientId && config.clientSecret && config.host
+    ? CognitoProvider({
+        id: config.id,
+        clientId: config.clientId,
+        clientSecret: config.clientSecret,
+        issuer: config.host,
+        name: config.name ?? DEFAULT_NAME,
+        authorization: {
+          params: {
+            scope: config.scope || 'openid email profile',
+          },
         },
-      },
-      token: tokenConfig,
-    }),
-];
+        token: tokenConfig,
+      })
+    : null;
+
+const getOktaProvider = (config: ProviderConfig) =>
+  config.clientId && config.clientSecret && config.issuer
+    ? OktaProvider({
+        id: config.id,
+        clientId: config.clientId,
+        clientSecret: config.clientSecret,
+        issuer: config.issuer,
+        authorization: {
+          params: {
+            scope: config.scope || 'openid email profile',
+          },
+        },
+        token: tokenConfig,
+      })
+    : null;
+
+const getProviderFromConfig = (config: ProviderConfig) => {
+  switch (config.provider) {
+    case SupportedProviders.AUTH0:
+      return getAuth0Provider(config);
+    case SupportedProviders.GOOGLE:
+      return getGoogleProvider(config);
+    case SupportedProviders.AZURE_AD:
+      return getAzureProvider(config);
+    case SupportedProviders.AZURE_B2C:
+      return getAzureB2CProvider(config);
+    case SupportedProviders.GITLAB:
+      return getGitLabProvider(config);
+    case SupportedProviders.PING_ID:
+      return getPingIdProvider(config);
+    case SupportedProviders.KEYCLOAK:
+      return getKeycloakProvider(config);
+    case SupportedProviders.COGNITO:
+      return getCognitoProvider(config);
+    case SupportedProviders.OKTA:
+      return getOktaProvider(config);
+    default:
+      return null;
+  }
+};
+
+const getProviderEnv = (
+  provider: SupportedProviders,
+  envName: ProviderConfigFields,
+  index = 0,
+) => {
+  const indexStr = index ? `_${index}` : '';
+  return process.env[`AUTH_${provider}${indexStr}_${envName}`];
+};
+
+const providerNames = {
+  [SupportedProviders.AUTH0]: 'auth0',
+  [SupportedProviders.AZURE_B2C]: 'azureB2C',
+  [SupportedProviders.AZURE_AD]: 'azure',
+  [SupportedProviders.COGNITO]: 'cognito',
+  [SupportedProviders.GOOGLE]: 'google',
+  [SupportedProviders.KEYCLOAK]: 'keycloak',
+  [SupportedProviders.OKTA]: 'okta',
+  [SupportedProviders.GITLAB]: 'gitlab',
+  [SupportedProviders.PING_ID]: 'pingId',
+};
+
+const getProviderConfig = (provider: SupportedProviders, index = 0) => {
+  const config = {
```

Comment:
can be simplified to read
```suggestion
  const getEnv = (name:string) =>  getProviderEnv(provider, name, index);
  const config = {
```
and then
```
getEnv(ProviderConfigFields.CLIENT_ID),
getEnv(ProviderConfigFields.CLIENT_SECRET)
```